src/vec_int64_ppc.h

```diff
@@ -92,14 +100,752 @@
+ * - Scalar and Vector test data class
+ * - Permute and Permute right index
+ *
+ * An impressive list of operation that can be used for;
+ * - Vectorizing long integer loops
+ * - Implementing useful quadword integer operations which do not have
+ * corresponding PowerISA instructions
+ * - implementing extended precision multiply and multiplicative
+ * inverse operations
+ *
+ * The challenge is that useful operations available for POWER9 will
+ * need equivalent implementations for POWER8 and POWER7.  Similarly
+ * for operations introduced for POWER8 will need POWER7
+ * implementations.  Also there are some obvious missing operations;
+ * - Absolute Difference doubleword (we have  byte, halfword, and word)
+ * - Average doubleword (we have byte, halfword, and word)
+ * - Extend Sign doubleword to quadword (we have byte, halfword, and word)
+ * - Multiply-sum word (we have byte, halfword, and doubleword)
+ * - Multiply even/odd doublewords (we have byte, halfword, and word)
+ *
+ * \subsection i64_missing_ops_0_1 Challenges and opportunities
+ *
+ * The stated goals for pveclib are:
+ *  - Provide equivalent functions across versions of the compiler.
+ *  - Provide equivalent functions across versions of the PowerISA.
+ *  - Provide complete arithmetic operations across supported C types.
+ *
+ * So the first step is to provide implementations for the
+ * key POWER8 doubleword integer operations for older compiles.
+ * For example some of the generic doubleword integer operations
+ * were not defined until GCC 6.0.
+ * Here we define the specific Compare Equal Unsigned
+ * Doubleword implementation:
+ * \code
+static inline
+vb64_t
+vec_cmpequd (vui64_t a, vui64_t b)
+{
+  vb64_t result;
+#ifdef _ARCH_PWR8
+#if __GNUC__ >= 6
+  result = vec_cmpeq(a, b);
+#else
+  __asm__(
+      "vcmpequd %0,%1,%2;\n"
+      : "=v" (result)
+      : "v" (a),
+      "v" (b)
+      : );
+#endif
+#else
+  // _ARCH_PWR7 implementation ...
+#endif
+  return (result);
+}
+ * \endcode
+ * The implementation checks of the compile target is POWER8 then
+ * checks of the compiler is new enough to use the generic vector
+ * compare built-in. If the generic built-in is not defined in
+ * <altivec.h> then we provide the equivalent inline assembler.
+ *
+ * For POWER7 targets we don't have any vector compare doubleword
+ * operations and we need to define the equivalent operation using
+ * PowerISA 2.06B (and earlier) instructions.
+ * For example:
+ * \code
+#else
+  // _ARCH_PWR7 implementation ...
+  vui8_t permute =
+    { 0x04,0x05,0x06,0x07, 0x00,0x01,0x02,0x03,
+      0x0C,0x0D,0x0E,0x0F, 0x08,0x09,0x0A,0x0B};
+  vui32_t r, rr;
+  r = (vui32_t)vec_cmpeq((vui32_t)a, (vui32_t)b);
+  if (vec_any_ne((vui32_t)a, (vui32_t)b))
+    {
+       rr = vec_perm (r, r, permute);
+       r= vec_and (r, rr);
+    }
+  result = (vb64_t)r;
+#endif
+ * \endcode
+ * Here we use compare equal unsigned word. If all words are equal
```

Comment:
Did you want to capitalize "compare equal unsigned word" ?
Comma after "If all words are equal".